python_modules/dagster/dagster/__init__.py

```diff
@@ -253,45 +254,97 @@
     execute_solid_within_pipeline,
     execute_solids_within_pipeline,
 )
+from pep562 import pep562
 
 from .version import __version__
 
 from dagster.config.source import BoolSource, StringSource, IntSource  # isort:skip
 
-# DEPRECATIONS
-# TODO: find a way to add warnings
-EventMetadataEntry = MetadataEntry
-EventMetadata = MetadataValue
-TextMetadataEntryData = TextMetadataValue
-UrlMetadataEntryData = UrlMetadataValue
-PathMetadataEntryData = PathMetadataValue
-JsonMetadataEntryData = JsonMetadataValue
-MarkdownMetadataEntryData = MarkdownMetadataValue
-PythonArtifactMetadataEntryData = PythonArtifactMetadataValue
-FloatMetadataEntryData = FloatMetadataValue
-IntMetadataEntryData = IntMetadataValue
-DagsterPipelineRunMetadataEntryData = DagsterPipelineRunMetadataValue
-DagsterAssetMetadataEntryData = DagsterAssetMetadataValue
-TableMetadataEntryData = TableMetadataValue
-TableSchemaMetadataEntryData = TableSchemaMetadataValue
+# ########################
+# ##### DEPRECATED ALIASES
+# ########################
 
+# NOTE: Unfortunately we have to declare deprecated aliases twice-- the
+# TYPE_CHECKING declaration satisfies linters and type checkers, but the entry
+# in `_DEPRECATED` is required  for us to generate the deprecation warning.
+
+if typing.TYPE_CHECKING:
+    # pylint:disable=reimported
+    from dagster.core.definitions import (
+        MetadataEntry as EventMetadataEntry,
+        MetadataValue as EventMetadata,
+        TextMetadataValue as TextMetadataEntryData,
+        UrlMetadataValue as UrlMetadataEntryData,
+        PathMetadataValue as PathMetadataEntryData,
+        JsonMetadataValue as JsonMetadataEntryData,
+        MarkdownMetadataValue as MarkdownMetadataEntryData,
+        PythonArtifactMetadataValue as PythonArtifactMetadataEntryData,
+        FloatMetadataValue as FloatMetadataEntryData,
+        IntMetadataValue as IntMetadataEntryData,
+        DagsterPipelineRunMetadataValue as DagsterPipelineRunMetadataEntryData,
+        DagsterAssetMetadataValue as DagsterAssetMetadataEntryData,
+        TableMetadataValue as TableMetadataEntryData,
+        TableSchemaMetadataValue as TableSchemaMetadataEntryData,
+    )
+
+    # pylint:enable=reimported
+
+_DEPRECATED = {
+    "EventMetadataEntry": ("MetadataEntry", MetadataEntry, "0.15.0"),
```

Comment:
not sure if i missed a comment in the other PR but the the class name, class duping in this tuple feels unnecessary

Reply:
good point, fixed it